universe/supplycommit/README.md

```diff
@@ -0,0 +1,135 @@
+# Universe Supply Commitment State Machine
+
+This package implements a state machine responsible for managing the on-chain
+commitment to the supply of a specific Taproot Asset (identified by its
+`asset.Specifier`, namely the group key).
+
+## Rationale and Purpose
+
+Taproot Assets allow for issuance and transfer off-chain, but maintaining a
+verifiable, global view of the *total supply* requires an on-chain commitment
+mechanism. This state machine addresses that need.
+
+Its primary purpose is to:
+
+1.  **Track Supply Changes:** Receive events representing new asset mints
+    (`NewMintEvent`), asset burns (`NewBurnEvent`), and ignored outputs
+    (`NewIgnoreEvent`).
+
+2.  **Batch Updates:** Collect these supply-altering events over time.
+
+3.  **Commit Periodically:** On a regular trigger (`CommitTickEvent`), create a
+    new cryptographic commitment (an MS-SMT root) representing the current state
+    of the asset's supply. This involves separate sub-trees for mints, burns, and
+    ignores, which are then committed into a single root supply tree.
+
+4.  **Anchor On-Chain:** Construct, fund, sign, and broadcast a Bitcoin
+    transaction that anchors this new supply root commitment into the
+    blockchain. This transaction spends the previous commitment output (if one
```

Comment:
So the way I understand the flow proposed in the current iteration is that we can only spend pre-commitment outputs from minting transactions that have already confirmed?
Which would mean we would either need to block the user from creating two mint transactions for the same group in the same block (to make sure there's a "linear" transaction chain), or allow one commitment transaction to spend multiple pre-commitment outputs.

If I understand this sentence correctly, we're aiming for the second option, is that right?

Reply:
> or allow one commitment transaction to spend multiple pre-commitment outputs.
> If I understand this sentence correctly, we're aiming for the second option, is that right?

Correct. Each time we go to make a new commitment, we'll query for _all_ the unspent pre-commitment outputs to spend in the next commitment transaction.